core/src/main/java/com/linecorp/armeria/server/ExceptionStats.java

```diff
@@ -0,0 +1,69 @@
+package com.linecorp.armeria.server;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.linecorp.armeria.common.util.Exceptions;
+import com.linecorp.armeria.internal.common.util.ReentrantShortLock;
+
+final class ExceptionStats {
+    private Map<String, ExceptionContext> exceptions;
+    private final ReentrantShortLock lock;
+
+    public ExceptionStats(Map<String, ExceptionContext> exceptions, ReentrantShortLock lock) {
+        this.exceptions = exceptions;
+        this.lock = lock;
+    }
```

Comment:
What do you think about:

- Making the constructor package private, because it doesn't make sense for a package-private class to have a public constructor; and
- Replacing the constructor with the default constructor so we instantiate `ConcurrentHashMap` and `ReentrantShortLock` here, because we always call this constructor with the same parameters.